integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/altinn/MessagePolling.java

```diff
@@ -135,44 +133,48 @@ public void checkForNewMessages() throws MessageException {
         }
 
         for (FileReference reference : fileReferences) {
-            final DownloadRequest request = new DownloadRequest(reference.getValue(), properties.getOrg().getNumber());
-            EduDocument eduDocument = client.download(request);
+            try {
+                final DownloadRequest request = new DownloadRequest(reference.getValue(), properties.getOrg().getNumber());
+                EduDocument eduDocument = client.download(request);
+
+                if (isNextMove(eduDocument)) {
+                    log.info("NextBest Message received");
+                    client.confirmDownload(request);
+                    Audit.info("Message downloaded", markerFrom(reference).and(eduDocument.createLogstashMarkers()));
+                    if (!properties.getNoarkSystem().getEndpointURL().isEmpty()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS134)